rasa/model.py

```diff
@@ -22,25 +23,75 @@
 if typing.TYPE_CHECKING:
     from rasa.importers.importer import TrainingDataImporter
 
-# Type alias for the fingerprint
-Fingerprint = Dict[Text, Union[Text, List[Text], int, float]]
 
 logger = logging.getLogger(__name__)
 
+
+# Type alias for the fingerprint
+Fingerprint = Dict[Text, Union[Text, List[Text], int, float]]
+
 FINGERPRINT_FILE_PATH = "fingerprint.json"
 
 FINGERPRINT_CONFIG_KEY = "config"
 FINGERPRINT_CONFIG_CORE_KEY = "core-config"
 FINGERPRINT_CONFIG_NLU_KEY = "nlu-config"
-FINGERPRINT_DOMAIN_KEY = "domain"
+FINGERPRINT_DOMAIN_WITHOUT_NLG_KEY = "domain"
+FINGERPRINT_NLG_KEY = "nlg"
 FINGERPRINT_RASA_VERSION_KEY = "version"
 FINGERPRINT_STORIES_KEY = "stories"
 FINGERPRINT_NLU_DATA_KEY = "messages"
 FINGERPRINT_TRAINED_AT_KEY = "trained_at"
 
 
+Section = namedtuple("Section", ["name", "relevant_keys"])
+
+SECTION_CORE = Section(
+    name="Core model",
+    relevant_keys=[
+        FINGERPRINT_CONFIG_KEY,
+        FINGERPRINT_CONFIG_CORE_KEY,
+        FINGERPRINT_DOMAIN_WITHOUT_NLG_KEY,
+        FINGERPRINT_STORIES_KEY,
+        FINGERPRINT_RASA_VERSION_KEY,
+    ],
+)
+SECTION_NLU = Section(
+    name="NLU model",
+    relevant_keys=[
+        FINGERPRINT_CONFIG_KEY,
+        FINGERPRINT_CONFIG_NLU_KEY,
+        FINGERPRINT_NLU_DATA_KEY,
+        FINGERPRINT_RASA_VERSION_KEY,
+    ],
+)
+SECTION_NLG = Section(name="NLG Templates", relevant_keys=[FINGERPRINT_NLG_KEY])
+
+
+class ShouldRetrain(object):
```

Comment:
there are only newstyle classes in Python 3
```suggestion
class ShouldRetrain:
```